git.md

```diff
@@ -1,2 +1,127 @@
+
+## 5. Explain how to test the effect of the changes you made
+
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
+
+## 6. Structure each commit message like an email
+
+Like an email, commit messages have a subject and an optional body. The subject
+should start with a line summarizing your changes (aim for 50 characters or
+less). Use the imperative, present tense -- for example, "Fix typo",
+not "Fixed typo" or "Fixes typo". The message body, if present,
+should follow a blank line; wrap around 70 characters; and explain the
+motivation for the change ("why", not "what" or "how").
+
+See also:
+
+* [Example of a commit message](https://github.com/2DegreesInvesting/resources/issues/74).
+* <https://r-pkgs.org/git.html#commit-best-practices.>
+
+## 7. Agree on a workflow
+
+Choose a workflow based on your team's needs and preferences. However you
+choose to work, just make sure to agree on a common workflow that everyone
+follows:
+
+GitHub workflow (a.k.a Git feature branch workflow:
+
+* Uses a central repository. 
+* Uses a single long-lived branch: "main"; it never contains broken code.
+* Developers work on "feature branches, share them via pull requests, and
+peers review the changes before they become a part of the main codebase.
+* A new release is created by tagging a particular commit on the main branch.
+
+Gitflow workflow:
+
+* Like the GitHub workflow but uses two long-lived branches: "main"
+and "develop". The main branch stores the official release history,
+and the develop branch serves as an integration branch for features.
+
+See also: <https://www.atlassian.com/git/tutorials/comparing-workflows>.
+
+## 8. Use tools to be more productive
+
+Choose the tools that help you work more comfortably, more productively,
+and with fewer errors.
+
+See also:
+
+* [Install a Git client](https://happygitwithr.com/git-client.html)
+* [oh-my-zsh: Unleash your terminal like never before](https://ohmyz.sh/).
+* [gh: Take GitHub to the command line](https://cli.github.com/).
+* [Git from RStudio](https://rstudio.com/resources/webinars/managing-part-2-github-and-rstudio/).
+
+## References
+
+[Version control best practices](https://www.git-tower.com/blog/version-control-best-practices/)
+
+## How
+
+* [Git practices by example](TODO)
+* <https://github.com/2DegreesInvesting/coding-helpdesk#coding-helpdesk>.
+* <https://github.com/git-guides/>.
+* <https://coderefinery.github.io/git-intro/>.
+* <https://bit.ly/book-git-in-practice (Part 4: Git best practices)>.  
+* <https://dev.to/lydiahallie/cs-visualized-useful-git-commands-37p1>.
+
+## Discuss
+
+* Jackson: Always include files like LICENSE, README.md, and consider
+.gitignore and CONTRIBUTING.md.  
```

Comment:
@jdhoffa, this seems a bit out of scope for the Git theme (except maybe .gitignore). Should this move to another theme, maybe "Languagge agnostic practices"?